sympy/solvers/ode.py

```diff
@@ -4003,6 +4012,42 @@ def _frobenius(n, m, p0, q0, p, q, x0, x, c, check=None):
 
     return frobdict
 
+def _check_substitution_type_match(eq, func):
+    r"""
+    Matches any differential equation that `order_reducing_substitution` can solve.
+    For this to work equation should have a minimum order of derivative to be 1, i.e.,
+    `f(x)` should not be present.
+    """
+    x = func.args[0]
+    f = func.func
+    order_to_subs = 0
+    D = Dummy()
+    for nsubs_try in range(1,ode_order(eq,f(x))):
+        if eq.subs(f(x).diff(x, nsubs_try), D).has(f(x)):
+            break
+        else:
+            order_to_subs += 1
+    if order_to_subs > 0:
+        return {'var':order_to_subs, 'solutions':True}
+    else:
+        return {'var':order_to_subs, 'solutions':False}
+
+# Use repeated substitution until we do not have a function independent of derivative
+def ode_order_reducing_substitution(eq, func, order, match):
+    r"""
+    Substitutes lowest order derivate in equation to function with order
+    of derivative as 0.Eg `f^(n)(x) = g(x)`, where n is the least order derivate.
+    match[var] or n here is how many times the function if solved is to be integrated
+    to  get f(x) since `g(x) = f^(match[var])(x)`.
+    """
+    x = func.args[0]
+    f = func.func
+    g = Function('g')
+    eq = eq.subs(f(x).diff(x, match['var']), g(x))
```

Comment:
Maybe .atoms(Derivative ) will do.

Reply:
There are a lot of possible improvements that can be done like I suggested about ricatti equation above and many people have asked to truncate the series and then evaluated.Once that will be done it will solve many other problems.

About the implementation:===>

But why would someone add arbitrary function?
Otherwise I think maybe looping through all .atoms(Function) and defining a function min_order would be good.

Reply:
Their equation to solve may have an arbitrary function in it like `g(x)*f(x).diff(x, 2) + f(x).diff(x)`.